services/motion/builtin/move_request.go

```diff
@@ -229,16 +232,37 @@ func (ms *builtIn) newMoveOnGlobeRequest(
 	if !ok {
 		return nil, resource.DependencyNotFoundError(movementSensorName)
 	}
+	before := time.Now()
+	// NOTE: This is the location of the robot, as repoorted by the GPS
+	// at the beginning of the MoveOnGlobe call.
 	origin, _, err := movementSensor.Position(ctx, nil)
+	ms.logger.Debugf("move on globe origin: %v", origin)
+	validatedExtra.extra["origin"] = origin
+	after := time.Now()
 	if err != nil {
 		return nil, err
 	}
 
+	if err := motion.LogGP(*origin, before, after); err != nil {
+		return nil, err
+	}
+
+	if err := motion.ValidateGeopoint(origin); err != nil {
+		return nil, err
+	}
+
 	// add an offset between the movement sensor and the base if it is applicable
+	// NOTE: Why do we create the baseOrigin as a pose in frame with a zero pose?
 	baseOrigin := referenceframe.NewPoseInFrame(componentName.ShortName(), spatialmath.NewZeroPose())
+	// NOTE: I don't understand what a TransformPose is. I see that we pass in the base origin, the movement sensor's name & nothing else
+	// NOTE: I don't know what it measn to transform a pose in that context.
+	// NOTE: I don't understand under what conditions this would fail & why we should still execute MoveOnGlobe if it fails
```

Comment:
This would fail if the movement sensor is not in the framesystem. If this information is provided we can account for mountpoints, e.g. if the lidar producing a SLAM map is not precisely centered on a base. But in practice the divergence is usually sufficiently small that we can ignore and assume they're coincident and not require the information.